api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -70,6 +76,20 @@ public static Type fromTypeName(String typeString) {
       return TYPES.get(lowerTypeString);
     }
 
+    if (lowerTypeString.startsWith("geometry")) {
+      Matcher geometry = GEOMETRY_PARAMETERS.matcher(typeString.substring(8));
+      if (geometry.matches()) {
+        return GeometryType.of(geometry.group(1));
+      }
+    }
+
+    if (lowerTypeString.startsWith("geography")) {
+      Matcher geography = GEOGRAPHY_PARAMETERS.matcher(typeString.substring(9));
+      if (geography.matches()) {
+        return GeographyType.of(geography.group(1), geography.group(2));
```

Comment:
I think that this should validate the algorithm and pass the algorithm enum symbol instead of a string. That avoids being able to compile with a typo (because `EdgeInterpolationAlgorithm.VINCENTI` is not a symbol) instead of allowing code paths that contain `"vincenti"`.

Reply:
Validated algorithm here.